### ISSUE TITLE
Type definition of useFormState requires index signature

### ISSUE THREAD

**Issue by emlai** [wsmd/react-use-form-state#22]
With the following:

```ts

interface ConnectFormState {
  user: string
  host: string
  password: string
  database: string
  port: number
}

const form = useFormState<ConnectFormState>()
```
I get:
```
TS2344: Type 'ConnectFormState' does not satisfy the constraint '{ [key: string]: string | number | string[]; }'.
  Index signature is missing in type 'ConnectFormState'.
```

I don't want to add an index signature to my form state.

Changing `useFormState` definition to the following seems to fix this:

```ts
export function useFormState<
  // old: T extends { [key: string]: string | string[] | number }
  T extends { [key in keyof T]: string | string[] | number }
>(
```

And it still disallows properties not of type `string | string[] | number` e.g. `boolean`s.

**Comment by wsmd** [comment]
Hi @emlai! Thanks for reporting this issue.

> I don't want to add an index signature to my form state.

We definitely don't want that! 😱

I'll push a fix shortly.

**Comment by wsmd** [comment]
This is now fixed in the latest release `0.6.1`.

`boolean`s are now allowed as well.

Please keep in mind that even though `useFormState` accepts numeric values in the initial state, the input value returned in the form state for numeric inputs (e.g. `input.number` and `input.range`) will be of type `string`.

```ts
interface ConnectFormState {
  user: string;
  remember: boolean,
  port: number;
}

const [state] = useFormState<ConnectFormState>({
  port: 8080
});

const port: string = state.port; // "8080"
```

This _may_ change in upcoming releases.

Thanks again!
